AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/Support/MutationEvent+Extensions.swift

```diff
@@ -30,21 +29,34 @@ extension MutationEvent {
                     return
                 }
 
-                if existingEvent.version == nil {
-                    Amplify.log.verbose("""
-                        Replacing existing mutation event having nil version with version from mutation response
-                            \(mutationSync.syncMetadata.version)
-                        """)
-                    existingEvent.version = mutationSync.syncMetadata.version
-                    storageAdapter.save(existingEvent, condition: nil) { result in
-                        switch result {
-                        case .failure(let dataStoreError):
-                            completion(.failure(dataStoreError))
-                        case .success:
-                            completion(.success(()))
+                // check if version returned from API is not nil and <= version sent in the request
+                if existingEvent.version != nil && mutationSync.syncMetadata.version <= existingEvent.version! {
```

Comment:
1. if the `mutationSync.syncMetadata.version` is == to `existingEvent.version` then if the model equality check checks out, we replace it anyways- so i think it should just execute the below logic for `<`.
2. if `existing.version == nil` it continues to the logic below. before it would replace it regardless of what the response version is, but now we check the model. do we want to do this? if the existing.version is `nil`, can we avoid the model check and simply replace it?

Reply:
`if existingEvent.version!=nil && existingEvent.version >= mutationSync.syncMetadata.version`